Duplicati/Library/Main/Controller.cs

```diff
@@ -671,21 +671,18 @@ private void SetupCommonOptions(ISetCommonOptions result, ref string[] paths, re
             foreach (Library.Interface.IGenericModule m in DynamicLoader.GenericLoader.Modules)
                 m_options.LoadedModules.Add(new KeyValuePair<bool, Library.Interface.IGenericModule>(Array.IndexOf<string>(m_options.DisableModules, m.Key.ToLower()) < 0 && (m.LoadAsDefault || Array.IndexOf<string>(m_options.EnableModules, m.Key.ToLower()) >= 0), m));
 
-            var conopts = new Dictionary<string, string>(m_options.RawOptions);
+            // Make the filter read-n-write able in the generic modules
+            var pristinefilter = string.Join(System.IO.Path.PathSeparator.ToString(), FilterExpression.Serialize(filter));
+            m_options.RawOptions["filter"] = pristinefilter;
+            
             var qp = new Library.Utility.Uri(m_backend).QueryParameters;
             foreach(var k in qp.Keys)
-                conopts[(string)k] = qp[(string)k];
-
-            // Make the filter read-n-write able in the generic modules
-            var pristinefilter = conopts["filter"] = string.Join(System.IO.Path.PathSeparator.ToString(), FilterExpression.Serialize(filter));
+                m_options.RawOptions[(string)k] = qp[(string)k];

```

Comment:
Hmm, good question...
Generally, the code should not change the m_options dictionary, except for the special case where user-provided code is added (the script module).

The only class currently implementing the `IConnectionOptions` interface is `HttpOptions`. It needs the URL stuff because it could include allowed certificate hashes etc., but it does not need to modify the options.

I think the updated version achieves the same as before, but with the script problem fixed.

As for not exposing the password, I agree we should fix that, but it is currently leaking from other places as well (into the WebUI). We should revisit that idea once we shield it from the WebUI.

The other question "...letting URL options override command line options always...".
I would find it surprising if I pasted a URL, and that changed some of my non-connection settings. I don't think we can use just "override" because options have default values, and having the non-overriding side set it can change the value, except if it is also set by the overriding side. 
With the current setup, the url-parameters always override the commandline options, but the options are only seen in the backend (and the `HttpModule`).